### PR TITLE
chore: add change detection rule and update components

### PR DESCRIPTION
**Pull request by amt-augustineopokukesse**
- Added on push change detection rule to linting rules
- Changed components change detection rule to on push

### CODE REVIEW

File: apps/iglidur-designer/src/app/libs/header/src/lib/header/header.component.ts [hunk 1]
```diff
@@ -53,6 +55,7 @@ export class HeaderComponent implements OnInit, OnDestroy {
       .subscribe((language) => {
         this.language = language;
         this.translate.use('header.component.i18n');
+        this.changeDetectorRef.markForCheck();
```
Comment (Bilal-Abubakari) [inline apps/iglidur-designer/src/app/libs/header/src/lib/header/header.component.ts:58]: Are you sure this is necessary?

Comment (amt-augustineopokukesse) [reply apps/iglidur-designer/src/app/libs/header/src/lib/header/header.component.ts:58]: It still works without me adding it. But I read a resource that recommended adding it.
I'll remove it then.

Comment (Bilal-Abubakari) [reply apps/iglidur-designer/src/app/libs/header/src/lib/header/header.component.ts]: Yes you need it if you really need to notify angular that there is a change that should be reflected in the view  but in this case you are subscribing on initialization and keeping the stream open which make it reactive :)

**Review by Bilal-Abubakari — APPROVED** [review]

File: apps/iglidur-designer/src/app/libs/header/src/lib/header/header.component.ts (hunk 1 above)
Comment (amt-augustineopokukesse) [reply apps/iglidur-designer/src/app/libs/header/src/lib/header/header.component.ts]: Great.. thanks for the explanation

**Pull request merged**